src/smf/packet-smf-compress.c

```diff
@@ -74,14 +79,26 @@ typedef struct _smf_compressed_conv_t {
     smf_compressed_stream_t stream2;
 } smf_compressed_conv_t;
 
+static voidpf stream_alloc(void * opaque, uInt num, uInt size)
+{
+    (void)opaque; // avoid compiler warning
+    return wmem_alloc0(wmem_file_scope(), num*size);
+}
+
+static void stream_free(void * opaque, voidpf address)
+{
+    (void)opaque; // avoid compiler warning
+    wmem_free(wmem_file_scope(), address);
```

Comment:
Why do we need to use wmem?  Was there evidence of a memory leak if we just use the zlib default allocators?  If there is a memory leak there, isn't this more likely due to not calling inflateEnd() when we're done?

Reply:
The conversation is scoped by pcap (in each pcap, there could be multiple conversations. Each pcap can potentially be listening to multiple tcp exchanges, each exchange has two direction of data, each direction requires a  z_stream).
In one conversation, we keep z_stream for each direction. It is file() scoped because we want to keep it around for the first-pass in the wireshark scan. Once the first-pass is done, we can potentially remove the z_stream structure.

Reply:
This is the information that would clarify the use of the conversation structure. They are found in the file README.dissector.
The "data" below talks about the conversation data that dissect_smf_compressed() allocated.
Section 2.2.1.8 The conversation_add_proto_data function.... "data" is a pointer to the data you wish to associate with the conversation. "data" usually points to "wmem_alloc'd" memory; the
memory will be automatically freed each time a new dissection begins
and thus need not be managed (freed) by the dissector.

At the bottom of the README.dissector is a sample route on how to create the conversation data. In there, they use...
    data_ptr = wmem_alloc(wmem_file_scope(), sizeof(my_entry_t));